src/Device/deviceSlice.ts

```diff
@@ -123,11 +127,50 @@ const slice = createSlice({
         },
 
         addDevice: (state, action: PayloadAction<Device>) => {
-            state.devices.set(action.payload.serialNumber, {
+            const device = {
                 ...action.payload,
                 favorite: getPersistedIsFavorite(action.payload.serialNumber),
                 nickname: getPersistedNickname(action.payload.serialNumber),
-            });
+            };
+
+            const persistedSerialPortOptions = getPersistedSerialPortOptions(
+                action.payload.serialNumber
+            );
+
+            if (persistedSerialPortOptions) {
+                const path =
+                    action.payload.serialPorts?.[
+                        persistedSerialPortOptions.vComIndex
+                    ].comName;
+
+                if (path) {
+                    device.persistedSerialPortOptions = {
+                        ...persistedSerialPortOptions.serialPortOptions,
+                        path,
+                    };
+                }
+            }
+
+            state.devices.set(action.payload.serialNumber, device);
+        },
+
+        persistSerialPortOptions: (
+            state,
+            action: PayloadAction<SerialPortOpenOptions<AutoDetectTypes>>
+        ) => {
+            if (state.selectedSerialNumber === null) return;
+
+            const vComIndex = state.devices
+                .get(state.selectedSerialNumber)
+                ?.serialPorts?.findIndex(e => e.path === action.payload.path);
+
+            if (vComIndex) {
```

Comment:
if vComIndex is 0 this will fail 

consider if (vComIndex !== undefined)

Reply:
Or `vComIndex >= 0` 🤷